### ISSUE TITLE
Implement data cleaning part 2

### ISSUE THREAD

**Issue by devowit** [usc-isi-i2/t2wml#259]
A continuation of issue #159, after all the cleaning functions have been implemented, next steps:

1. support for a field in the yaml, to create a cleaned copy of the data file with cleaning functions applied to regions. 
(the name of the cleaned file will be ORIGINALNAME_cleaned.csv/xlsx/xls)

```
cleaning:
       - region:
                  columns/rows/range/cells/whatever
         functions:
                  - ftfy
                  - replace_regex
       - region
                    etc....
```

2. this file will be the one used in t2wml calculations. 

it will be possible to view the cleaned or the original file in the frontend (#260)

**Comment by devowit** [comment]
proposal for general structure of the yaml file:
```

statementMapping:
  region: 
    - range: D6:K20
      skip_cells:
        - =value[$col, $row] == " .."
  template:
    item: '=get_item(regex(value[B, 2], "profile: (.*) \d{4}", 1))' 
    property: P1082
    value: =replace_regex(value[$col, $row], "[^\d.-]", "")
cleaningMapping:
  - region: 
         - range: D6:K20
           skip_cells:
                 - =value[$col, $row] == " .."
     functions:
        - ftfy
        - strip_whitespace:
                char: "\t"
               where: end
 - region:  
          - left: D
            right: F
            top: 4
            bottom: 9
   functions:
          - strip_whitespace # default
```

**Comment by devowit** [comment]
Here is a sample yaml, just the cleaningMapping section, which includes all the functions and their possible arguments (as required arguments have been left blank, this isn't a _valid_ yaml unless those get filled in.

```
cleaningMapping:
       - region:
            - range: D6:K20 #or any other valid region arguments
         functions:
            - ftfy
            - strip_whitespace:
                char: null # default all whitespace, can also be " " or  "\t"
                where: start_and_end
            - replace_regex:
                to_replace: #required, no default
                replacement: #required, no default
            - remove_numbers:
                where: everywhere
            - remove_letters:
                where: everywhere
            - truncate:
                length: #required, no default
            - normalize_whitespace:
                tab: False
            - change_case:
                case: "sentence" #can also be "lower", "upper", and "title"
            - pad:
                length: #required, no default
                pad_text: #required, no default
                where: start # or "end". does not allow "everywhere" or "start_and_end"
            - make_numeric:
                decimal: "."
            - make_alphanumeric
            - make_ascii:
                translate: False
```

**Comment by devowit** [comment]
released in 2.3.9, bugs and additional feature requests should be opened as separate issues
